src/motion_input_shaping/shaped_axis.py

```diff
@@ -222,6 +366,10 @@ def move_min_shaped(
 
 # Example code for using the class.
```

Comment:
IMO some simple example code is nice, but Bruce does raise a good point. The vast majority of the code from this file is copied/pasted in shaped_lockstep.py. I think there should be a more elegant way to combine it. I have a feeling once we switch to composition instead of inheritance it will become more obvious. This may also be a weakness of ZML that a lockstep group can't be treated similar to a single axis (even though many of the methods are identical).

Reply:
I agree that simple examples right in the class files are nice.

For this particular case, after the refactoring, how different is this piece of test code between `shaped_axis.py` and `shaped_axis_stream.py`?  If they are really similar, does it make sense to split them out in a separate file named `<something>_test.py` like the other files that we want users to run?

Reply:
I have removed the example code from these files and created `shaping_demo.py` which does a simple demo of both classes and the methods in them.

I have also done a few changes to `shaping_comparison_test.py` so that is performs and plots shaping with both shaping classes so that they can be compared easily. In the plotting I added a subplot (the upper plot) to show the full trajectory of without zooming in so that it is easy to see the trajectory and the delay from communication and the device response on bottom that is zoomed in to see residual vibrations. An example is shown below but the device I used doesn't have a large oscillation so the response is not obvious.

![image](https://github.com/zabertech/zaber-examples/assets/144398940/52c34809-8ea7-4431-accc-35af91dc9816)